### PR TITLE
Implement IFrame.LoadRequest with post data

### PR DESCRIPTION
**Pull request by rover886**
This an extended work depends on pull request
https://github.com/cefsharp/CefSharp/pull/1440. In this pull request
(#1440) @amaitland enabled `LoadRequest` method of IFrame, I worked on
remaining part to add post data.

### CODE REVIEW

**Comment by amaitland** [comment]
@rover886 Can you fix the formatting first, spaces instead of tabs. If you view the `Files changed` above you'll see the problem with formatting.

Is there a particular reason you've targeted this against the `43` branch? I usually prefer new features be targeted against `master`.

**Comment by rover886** [comment]
@amaitland I am extremely sorry about bad formatting, actually in VS 2013 I can see it very properly, but still I am sorry for not following pull request and commit guidelines. It's my first time to create pull request and commit on GitHub so kinda...you know...I have mess up all the things :( :-1:

**Comment by amaitland** [comment]
@rover886 It's no big deal. There are a few relatively easy options for fixing the formatting, try something like http://blogs.msdn.com/b/zainnab/archive/2010/03/14/how-to-convert-tabs-to-spaces-and-vice-versa-vstipedit0028.aspx

It appears your build is failing as well see https://ci.appveyor.com/project/cefsharp/cefsharp/build/43.0.0-CI1603

**Comment by rover886** [comment]
> It appears your build is failing as well 

In VS 2013 I re-confirmed build is successful :(

**Comment by amaitland** [comment]
> In VS 2013 I re-confirmed build is successful :(

The `43` branch is compiled using `VS2012`, so it's likely you've used a language feature that's not supported in `VC++ 2012`.

**Comment by rover886** [comment]
>so it's likely you've used a language feature that's not supported in VC++ 2012.

or Is it like I have added two new files:

CefSharp.Core/Internals/CefPostDataElementWrapper.cpp
CefSharp.Core/Internals/CefPostDataWrapper.cpp

But, I have not committed project `CefSharp.Core` file

File: CefSharp.Core/Internals/CefPostDataWrapper.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#include "Stdafx.h"
```
Comment (amaitland) [inline CefSharp.Core/Internals/CefPostDataWrapper.cpp]: Can you merge this back into the `.h` file? I know that sounds like a strange request, It just makes comparing your changes so much easier.

Comment (amaitland) [inline CefSharp.Core/Internals/CefPostDataWrapper.cpp]: As a side note, when adding a new file, need to add license disclaimer.

**Comment by amaitland** [comment]
> CefSharp.Core/Internals/CefPostDataElementWrapper.cpp
> CefSharp.Core/Internals/CefPostDataWrapper.cpp
> 
> But, I have not committed project CefSharp.Core file

Yep, that's likely it.

**Comment by amaitland** [comment]
As I commented inline, I'd prefer if you merged the `.cpp` content into their respective `.h` files, makes it easier to compare your code.

**Comment by rover886** [comment]
@amaitland I mess up everything on such a nice project :(

I want to take this pull request down :-1: , So I will commit everything with nice formatting. Is their any way to take pull request down?

Also is there any way to see my changes online (on website not in Github desktop) before creating pull request? If no, then sorry to say but, I don't want to create any pull request :(.. I mess up everything on best project. Sorry again.

**Comment by amaitland** [comment]
> I want to take this pull request down :-1: , So I will commit everything with nice formatting. Is their any way to take pull request down?

No need. It's not a big deal, it's just formatting. Just needs to be fixed before merging. You can of course squash your commits in a single commit, then overrite existing changes on the branch, removing any trace of your old code. I'm not sure how to do that on `GitHub` desktop, as I've not used it for a long time.

> Also is there any way to see my changes online (on website not in Github desktop) before creating pull request? If no, then sorry to say but, I don't want to create any pull request :(.. I mess up everything on best project. Sorry again.

Don't give up now, won't take long to sort out :+1:

**Comment by amaitland** [comment]
If you want, I can squash all your commits into one before I merge :smile:

**Comment by rover886** [comment]
@amaitland I have committed CefSharp.`Core.vcxproj` how can I add this commit in this request :disappointed:

**Comment by rover886** [comment]
I have modified method `IPostData.AddElement`. Previously it's parameter was `IPostDataElement element`, I modified it to accept `key` and `value` and in respective `.cpp` converted these key-value pair to `array<Byte>`, to accept file in `request` another overloaded method is added which accepts `FileName`.

I also wish that it will be more readable and convenient if in `IPostData.Elements` (get method) we get list of `IPostDataElement`, in which each object of `CefPostDataElementWrapper` will have `key-value` pair.

Once again @amaitland sorry for whatever mess up I did, and for bad coding practice if any.

**Comment by rover886** [comment]
closed by mistake

**Comment by amaitland** [comment]
> I have modified method IPostData.AddElement. Previously it's parameter was IPostDataElement element, I modified it to accept key and value and in respective .cpp converted these key-value pair to array<Byte>, to accept file in request another overloaded method is added which accepts FileName.

@rover886 In general I'd prefer to keep the underlying `API` as close to the `CEF API`  as possible with extension methods to make using it easier. 

I'm going to close this `PR` as I've finished the basic implementation over at #1440. Please check it out and let me know what you think. You'll see that I've chosen to use a lot of smaller more granular commits, in general I find this approach easier for others to review. You'll see I've implemented a few extension methods to make life a little easier, if you can think of any generic methods that would be helpful then please contribute :+1:

**Pull request closed without merging**